common/src/main/java/io/netty/util/ConstantPool.java

```diff
@@ -55,52 +56,53 @@ public T valueOf(Class<?> firstNameComponent, String secondNameComponent) {
      * @param name the name of the {@link Constant}
      */
     public T valueOf(String name) {
-        T c;
+        checkNotNullAndNotEmpty(name);
+        return getOrCreate(name);
+    }
 
-        synchronized (constants) {
-            if (exists(name)) {
-                c = constants.get(name);
-            } else {
-                c = newInstance0(name);
+    /**
+     * Get existing constant by name or creates new one if not exists. Threadsafe
+     *
+     * @param name the name of the {@link Constant}
+     */
+    private T getOrCreate(String name) {
+        T constant = constants.get(name);
+        if (constant == null) {
+            final T tempConstant = newConstant(nextId(), name);
```

Comment:
@johnou maybe. I haven't yet dig enough to say for sure. I don't to bring any regression so changing only things I'm 100% sure.